src/AppInstallerRepositoryCore/SourceFactory.h

```diff
@@ -41,6 +41,10 @@ namespace AppInstaller::Repository
         // Return value indicates whether the action completed.
         virtual bool Remove(const SourceDetails& details, IProgressCallback& progress) = 0;
 
+        // Edit the source from the given details..
+        // Return value indicates whether the action completed.
+        virtual bool Edit(const SourceDetails& details, IProgressCallback& progress) = 0;

```

Comment:
Similar to the progress, I'm not sure that we would allow edits to make the kinds of changes that require the actual source to be involved.  This should be removed unless you have a strong case for a future edit scenario.